main/lsp/requests/rename.cc

```diff
@@ -91,6 +90,74 @@ core::SymbolRef findRootClassWithMethod(const core::GlobalState &gs, core::Symbo
     return root;
 }
 
+class UniqueSymbolQueue {
+public:
+    UniqueSymbolQueue() : iter(0) {}
+
+    bool tryEnqueue(core::SymbolRef s) {
+        auto insertResult = set.insert(s);
+        bool isNew = insertResult.second;
+        if (isNew) {
+            symbols.emplace_back(s);
+        }
+        return isNew;
+    }
+
+    core::SymbolRef pop() {
+        if (iter < symbols.size()) {
+            auto result = symbols[iter];
+            iter++;
+            return result;
+        }
+        return core::SymbolRef();
+    }
+
+private:
+    int iter;
+    vector<core::SymbolRef> symbols;
```

Comment:
You can actually make this a `std::deque` so you can push things on the end and pop things off the front (or vice-versa). Then you don't need `iter`.